static/js/components/CommentList.jsx

```diff
@@ -551,6 +629,9 @@ const CommentList = ({
       {permissions.indexOf("Comment") >= 0 && gcnEventID && (
         <CommentEntry addComment={addGcnEventComment} />
       )}
+      {permissions.indexOf("Comment") >= 0 && shift_id && (
+        <CommentEntry addComment={addShiftComment} />
+      )}
```

Comment:
Here but also above for gcEvent comments and source comments, you check if the user has the permission and if there is an obj_id, shift_id or gcnEventID, but what you should also check (and it is the most important) is the associatedRessourceType, otherwise you end up having that component duplicated on the page.